### DISCUSSION TITLE
How to attach container to an existing network using its id

### DISCUSSION THREAD

**Discussion by ratcashdev** [testcontainers/testcontainers-java#4567] [Q&A] [answered]
I'd like to attach a newly pulled container to an existing set of containers (started externally using docker compose) running on network 'docker_default'.
Please advice how could this be done. Thanks.

Or simply allow the compose module to attach to an existing (and running) compose 'project' started from outside and create the socat container inside it so that we are able to do port maps.

**Comment by kiview** [comment, marked as answer]
You can instantiate a new `Network` using the external id:
```java
Network n = new Network() {
    @Override
    String getId() {
        return "existingId" // use id of external network
    }

    @Override
    void close() throws Exception {

    }

    @Override
    Statement apply(Statement base, Description description) {
        return null
    }
}

container.withNetwork(n)
```

Since this is a question rather than an issue, I converted it to a GH discussion.

**Reply by ratcashdev** [reply]
Thank you for the quick help.
BTW: Is there a similar way to 'attach' `DockerComposeContainer` to an existing set of running containers based on the compose file?

**Reply by kiview** [reply]
Not really on the Testcontainers side, since `docker-compose` will ultimately create the containers. But you can specify an external network in the compose file if this helps?

**Reply by ratcashdev** [reply]
No, not really. I was looking for a way to reuse the same programmatic interface of the `DockerComposeContainer` regardless if the containers were started by JAVA or having a long lifecycle and started from outside - just attach, allow some command execution, log introspection, container name, image, configuration retrieval, attach the SoCat container to the running set of containers for dynamic network maps.

Also, it would be nice if it could  - at least conditionally - support `container_name`. I know this would make scaling impossible, but what if one does not need scaling?
